sdk/storage/Azure.Storage.Blobs.Cryptography/src/LengthLimitingStream.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Storage.Blobs.Specialized.Cryptography
+{
+    using System;
+    using System.IO;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// A read-only stream. Wraps a <see cref="Stream"/> with an upper limit on how many bytes can be read.
+    /// </summary>
+    internal class LengthLimitingStream : Stream
+    {
+        private readonly Stream _wrappedStream;
+        private long? _maxLength;
+        private long _bytesRead = 0;
+
+        public LengthLimitingStream(Stream wrappedStream, long? length = default)
+        {
+            this._wrappedStream = wrappedStream;
+            this._maxLength = length;
+        }
+
+        public override bool CanRead => this._wrappedStream.CanRead;
+
+        public override bool CanSeek => false;
+
+        public override bool CanWrite => false;
+
+        public override long Length => this._maxLength.HasValue ? this._maxLength.Value : this._wrappedStream.Length;
```

Comment:
Does `_wrappedStream.Length` work safely here?  Let's say I have a 20 byte stream, seek to position 10, then wrap it in a `LengthLimitingStream`.  Should the Length be 20 when I can only see 10 bytes of it?

Reply:
Modified to always return _bytesRead.